clang/lib/Sema/TreeTransform.h

```diff
@@ -2893,6 +2893,8 @@ class TreeTransform {
 
     CXXScopeSpec SS;
     SS.Adopt(QualifierLoc);
+    if (FirstQualifierInScope)
+      SS.setFoundFirstQualifierInScope(FirstQualifierInScope);
```

Comment:
It looks like adding 'FirstQualifierInScope' as a property to SS makes it so keeping and passing both around is redundant.